paths_cli/wizard/joke.py

```diff
@@ -0,0 +1,55 @@
+import random
+
+from paths_cli.wizard.tools import a_an
+
+_NAMES = ['Fred', 'Winston', 'Agnes', "Millicent", "Ophelia", "Laszlo",
+          "Werner"]
+_THINGS = ['pet spider', 'teddy bear', 'close friend', 'school teacher',
+           'iguana']
+_SPAWN = ['daughter', 'son', 'first-born']
+
+_MISC = [
+    "Named after its father, perhaps?",
+    "Isn't '{name}' also the name of a village in Tuscany?",
+    "Didn't Johnny Cash write a song about {a_an_thing} named '{name}'?",
+    "I think I'm going to start rapping as 'DJ {name}'.",
+    "It would also be a good name for a death metal band.",
+]
+
+def _joke1(name, obj_type):  # no-cov
+    return (f"I probably would have named it something like "
+            f"'{random.choice(_NAMES)}'.")
+
+def _joke2(name, obj_type):  # no-cov
+    thing = random.choice(_THINGS)
+    joke = (f"I had {a_an(thing)} {thing} named '{name}' "
+            f"when I was young.")
+    return joke
+
+def _joke3(name, obj_type):  # no-cov
+    return (f"I wanted to name my {random.choice(_SPAWN)} '{name}', but my "
+            f"wife wouldn't let me.")
+
+def _joke4(name, obj_type):  # no-cov
+    a_an_thing = a_an(obj_type) + f" {obj_type}"
+    return random.choice(_MISC).format(name=name, obj_type=obj_type,
+                                       a_an_thing=a_an_thing)
+
+def name_joke(name, obj_type):  # no-cov
+    rnd = random.random()
+    if rnd < 0.25:
+        joke = _joke1
+    elif rnd < 0.50:
+        joke = _joke2
+    elif rnd < 0.65:
+        joke = _joke3
+    else:
+        joke = _joke4
```

Comment:
this can be rewritten to use [`random.choices`](https://docs.python.org/3/library/random.html#random.choices) (makes possible later extension easier):
```suggestion
    jokes = [_joke1, _joke2, _joke3, _joke4]
    weights = [5, 5, 3, 7]
    joke = random.choices(jokes, weights=weights)[0]
```

Reply:
Thanks! I had forgotten about `random.choices` -- it's exactly what I wanted. This if/elif stuff had been annoying me.